packages/core/test/actorLogic.test.ts

```diff
@@ -982,4 +983,45 @@ describe('composable actor logic', () => {
       }
     });
   });
+
+  it('higher-level logic should work', () => {
```

Comment:
this fix is specific to machines so I'd include "machine" in  the test title
```suggestion
  it('higher-level logic wrapping a machine should be able to persist a snapshot', () => {
```